go/backend-example/oauth/proxy.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+/*
+This code is a simple example of how to run a backend server for OAuth connections.
+You can send your request to localhost:8001/<api_path>
+This code will receive the request, gather the OAuth token and then forward it to our servers.
+You will need to provide your ACCESS_TOKEN_URL, CLIENT_ID, CLIENT_SECRET
+You can obtain these values following this guide: https://developer.payments.jpmorgan.com/quick-start
+If you are hitting our APIs that require a digital signature or JWT then uncomment line 40 and provide your key.
+Note this is not production code and is supplied to get developers started
+*/
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"sample_authentication_code/pkg/samples"
+)
+
+func main() {
+	remote, err := url.Parse("https://api-mock.payments.jpmorgan.com")
+	if err != nil {
+		panic(err)
+	}
+	access_token_url := "https://id.payments.jpmorgan.com/am/oauth2/alpha/access_token"
+	client_id := "3988f709-a8f9-47c3-9d28-c9fcc466d5c0"
+	client_secret := "y_OAgjpmVcEL1vpPDI9SXjIkdUYWBXpl8UIWHIVIlRW45zsl2q0EjfL_ffGkUDn02X2_RDZWe1buriJPrTwmeA"
+	//digital_key := ""
+	handler := func(p *httputil.ReverseProxy) func(http.ResponseWriter, *http.Request) {
+		return func(w http.ResponseWriter, r *http.Request) {
+			log.Println(r.URL)
+			r.Host = remote.Host
+			// Access token generation. We are using the code from our sample authentication folder
+			accessToken, err := samples.GetAccessToken(access_token_url, client_id, client_secret)
+			if err != nil {
+				log.Printf("Error getting access token: %v", err)
+			}
+			w.Header().Set("Authorization", "Bearer "+accessToken)
+			// JWT/Digital Signature generation. This is required for some of our POST requests.
+			//modifyRequestBody(w, r, digital_key)
```

Comment:
could run the method if digital_key is not empty/nil etc?

then just need a comment above the digital_key variable to say comment in and add this if your are using ?

can we pass those values in with command line as env vars? then can use os env etc?

Reply:
good idea, updated